### PR TITLE
RHCLOUD-26616 Prepare Splunk connector implementation

### PR DESCRIPTION
**Pull request by gwenneg**
This PR contains preparation work for the `notifications-connector-splunk` implementation.

### CODE REVIEW

File: connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java [hunk 1]
```diff
@@ -24,7 +24,6 @@ public class TeamsRouteBuilder extends EngineToConnectorRouteBuilder {
     public void configureRoute() {
         from(direct(ENGINE_TO_CONNECTOR))
                 .routeId(connectorConfig.getConnectorName())
-                .removeHeaders(CAMEL_HTTP_HEADERS_PATTERN)
```
Comment (MikelAlejoBR) [inline connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java:27]: Don't we have to remove these headers to the same way we do it in the `EngineToConnectorRouterBuilder` class?

**Review by MikelAlejoBR — APPROVED** [review]
LGTM!

File: connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java (hunk 1 above)
Comment (gwenneg) [reply connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java:27]: The line removed here didn't do anything with the latest version of the code. It was needed a few months ago when `engine` sent notifications to the Teams pseudo-connector through REST.

Comment (gwenneg) [reply connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java:27]: Before this PR, the headers management was a bit messy. Here's how headers work now:
- headers coming from Kafka are removed in `EngineToConnectorRouterBuilder` before the connectors start building/sending notifications to external services
- headers sent to external services are removed in `OutgoingCloudEventBuilder` before the connectors start sending the notifications history to the engine through Kafka
- each connectors is still free to use headers with no restrictions between these two boundaries

Comment (gwenneg) [reply connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java:27]: image

Comment (MikelAlejoBR) [reply connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java:27]: I understand! Thank you very much for the detailed explanation!

**Pull request merged**